apps/mark-scan/backend/src/electrical_testing/background.ts

```diff
@@ -28,9 +67,178 @@ export async function cardReadLoop({
   }
 }
 
-export async function printAndScanLoop(): Promise<void> {
-  // eslint-disable-next-line no-constant-condition
-  while (true) {
-    await sleep(5000);
+function generateTestPdf() {
+  return new Promise<Buffer>((resolve, reject) => {
+    const chunks: Uint8Array[] = [];
+    const doc = new PdfDocument({
+      size: [
+        TEST_DOC.WIDTH_IN * TEST_DOC.DPI,
+        TEST_DOC.HEIGHT_IN * TEST_DOC.DPI,
+      ],
+    });
+
+    doc.on('data', (chunk) => chunks.push(chunk));
+
+    doc.on('end', () => {
+      const pdfBuffer = Buffer.concat(chunks);
+      resolve(pdfBuffer);
+    });
+
+    doc.on('error', reject);
+
+    doc
+      .fontSize(12)
+      .text(
+        `${'Sample document '.repeat(3)}${'\n'.repeat(
+          40
+        )}${'Sample document end '.repeat(3)}`
+      )
+      .end();
+  });
+}
+
+export async function printAndScanLoop({
+  workspace,
+  logger,
+  controller,
+}: ElectricalTestingServerContext): Promise<void> {
+  function setPaperHandlerStatusMessage(message: string) {
+    workspace.store.setElectricalTestingStatusMessage('paper-handler', message);
+  }
+
+  controller.signal.addEventListener('abort', () => {
+    const message = `Print and scan loop stopping. Reason: ${controller.signal.reason}`;
+    void logger.log(LogEventId.BackgroundTaskCancelled, 'system', {
+      message,
+    });
+
+    setPaperHandlerStatusMessage(message);
+  });
+
+  const testPdf = await generateTestPdf();
+  const outputDir = join(workspace.path, 'electrical-testing-output');
+  await fs.mkdir(outputDir, { recursive: true });
+
+  await fs.writeFile(
+    join(outputDir, 'test-document.pdf'),
+    new Uint8Array(testPdf.buffer, testPdf.byteOffset, testPdf.byteLength)
+  );
```

Comment:
Thought it might be useful for debugging to write the test document to output but can remove if not needed.